lib/resolveurl/plugins/upstream.py

```diff
@@ -15,25 +15,19 @@
     You should have received a copy of the GNU General Public License
     along with this program. If not, see <http://www.gnu.org/licenses/>.
 """
-
+from resolveurl.plugins.__resolve_generic__ import ResolveGeneric
 from resolveurl.plugins.lib import helpers
-from resolveurl import common
-from resolveurl.resolver import ResolveUrl, ResolverError
 
 
-class UpStreamResolver(ResolveUrl):
+class UpStreamResolver(ResolveGeneric):
     name = "upstream.to"
     domains = ['upstream.to']
     pattern = r'(?://|\.)(upstream\.to)/(?:embed-)?([0-9a-zA-Z]+)'
 
     def get_media_url(self, host, media_id):
-        web_url = self.get_url(host, media_id)
-        headers = {'User-Agent': common.FF_USER_AGENT}
-        html = self.net.http_GET(web_url, headers=headers).content
-        sources = helpers.scrape_sources(html)
-        if sources:
-            return helpers.pick_source(helpers.sort_sources_list(sources)) + helpers.append_headers(headers)
-        raise ResolverError('Video not found')
+        return helpers.get_media_url(self.get_url(host, media_id),
+                                     patterns=[r'''sources:\s*\[(?:{file:)?\s*"(?P<url>[^"]+)'''],
+                                     generic_patterns=False)
```

Comment:
You need to add `referer=False` as well, otherwise you will get "embed restricted to domain" errors on some embeds similar to gounlimited